### PR TITLE
fix(a2-986): handle parsing and displaying of dates in UK time

### PR DESCRIPTION
**Pull request by bryn-pins**
## Ticket Number

https://pins-ds.atlassian.net/browse/A2-986

## Description of change

- Standardises the display of dates from BO
- When parsing user inputted dates treats input as UK time before saving as UTC

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by edpackard-work — DISMISSED** [review]

File: packages/common/src/lib/format-date.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const { formatInTimeZone, zonedTimeToUtc, utcToZonedTime } = require('date-fns-tz');
+const { isValid } = require('date-fns');
+
+const ukTime = 'Europe/London';
```
Comment (beejjacobs) [inline packages/common/src/lib/format-date.js:4]: If I'm picky (which I usually am...), a better name would be `ukTimezone`

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js [hunk 2]
```diff
@@ -81,7 +84,7 @@ class DateQuestion extends Question {
 			const answerDateString = journey.response.answers[this.fieldName];
 
 			if (answerDateString) {
-				const answerDate = new Date(answerDateString);
+				const answerDate = convertUTCToUK(answerDateString);
 				day = `${answerDate.getDate()}`.slice(-2);
 				month = `${answerDate.getMonth() + 1}`.slice(-2);
 				year = `${answerDate.getFullYear()}`;
```
Comment (beejjacobs) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js:90]: Is this tested? the `get*` methods are local time (which for the server is UTC) - you might need to use `formatInTimeZone` for each element?

Comment (bryn-pins) [reply packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js:90]: Have added some tests, is that enough to prove it works, or have I misunderstood how convertUTCToUK works?

File: packages/common/src/lib/format-date.js (hunk 1 above)
Comment (bryn-pins) [reply packages/common/src/lib/format-date.js]: done

**Review by beejjacobs — APPROVED** [review]
Comments addressed, new code is clear and avoids issues with date objects being changed.

**Pull request merged**